pkg/pillar/types/downloadertypes.go

```diff
@@ -138,11 +138,16 @@ func (status *DownloaderStatus) ClearPendingStatus() {
 
 // HandleDownloadFail : Do Failure specific tasks
 func (status *DownloaderStatus) HandleDownloadFail(errStr string, retryTime time.Duration, cancelled bool) {
-	if retryTime != 0 && !cancelled {
-		errStr = fmt.Sprintf("Will retry in %v: %s",
-			retryTime, errStr)
+	errDescription := ErrorDescription{
+		Error: errStr,
 	}
-	status.SetErrorNow(errStr)
+	if !cancelled && retryTime != 0 {
+		errDescription.Error = fmt.Sprintf("Will retry in %v: %s", retryTime, errStr)
```

Comment:
I think this can now be simplified to just Error = errStr since we have the retryCondition.